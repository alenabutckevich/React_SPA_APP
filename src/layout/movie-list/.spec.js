import React from 'react';
import toJson from 'enzyme-to-json';
import MovieList from './movie-list';
import Movie from './movie';

const setup = propOverrides => {
    const { movies } = propOverrides;
    const props = Object.assign({
        movies: movies,
        genres: [{
            id:1, name:'genre1'
        }, {
            id:2, name:'genre2'
        }]
    }, propOverrides);

    const wrapper = shallow(<MovieList {...props}/>);

    return {
        wrapper,
        props
    }
}

describe('<MovieList />', () => {

    it('render movie list', () => {
        const movies = [{
            id:1, title:'title1', poster_path:'path1', genre_ids:[1], release_date:'2012-12-12'
        }, {
            id:2, title:'title2', poster_path:'path2', genre_ids:[2], release_date:'2012-12-12'
        }];
        const { wrapper } = setup({movies: movies});

        expect(wrapper.find('Movie')).toHaveLength(2);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('render empty movie list', () => {
        const { wrapper } = setup([]);

        expect(wrapper.find('Movie')).toHaveLength(0);
        expect(wrapper.find('.movie-list_empty').text()).toEqual('No films found');
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})