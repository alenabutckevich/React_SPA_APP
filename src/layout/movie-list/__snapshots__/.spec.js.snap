// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieList /> render empty movie list 1`] = `
<div
  className="movie-list movie-list_empty"
>
  <div>
    No films found
  </div>
</div>
`;

exports[`<MovieList /> render movie list 1`] = `
<div
  className="movie-list "
>
  <div>
    <Movie
      genre="genre1"
      id={1}
      image="path1"
      key="1"
      releaseDate="12-12-2012"
      title="title1"
    />
    <Movie
      genre="genre2"
      id={2}
      image="path2"
      key="2"
      releaseDate="12-12-2012"
      title="title2"
    />
  </div>
</div>
`;
